### PR TITLE
[WIP] feat($plugin): Add vue-cli VuePress plugin

### PR DESCRIPTION
**Pull request by kefranabg**
**Summary**

> vue-cli plugin for VuePress

**What it looks like for now :**

Capture d’écran 2019-10-08 à 19 20 57

Capture d’écran 2019-10-08 à 19 20 36

Capture d’écran 2019-11-05 à 11 44 31

**Steps to test it locally:**
TODO

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,5 +53,8 @@
     "hooks": {
       "pre-commit": "lint-staged"
     }
+  },
```
Comment (kefranabg) [inline package.json]: To remove

File: package.json [hunk 2]
```diff
@@ -53,5 +53,8 @@
...
+  "dependencies": {
```
Comment (kefranabg) [inline package.json]: To remove

File: packages/@vuepress/vue-cli-plugin-vuepress/generator/template/docs/_vuepress/config.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  title: '<%= title %>',
+  description: '<%= description %>',
+  themeConfig: {
```
Comment (kefranabg) [inline packages/@vuepress/vue-cli-plugin-vuepress/generator/template/docs/_vuepress/config.js]: To remove

File: packages/@vuepress/vue-cli-plugin-vuepress/ui/config/write.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+module.exports = async ({ api, prompts }) => {
+  const result = {}
+
+  result['themeConfig.nav'] = []
+
+  for (const prompt of prompts) {
+    result[prompt.id] = await api.getAnswer(prompt.id)
+  }
+
+  if (get(result, 'themeConfig.sidebar')) {
```
Comment (kefranabg) [inline packages/@vuepress/vue-cli-plugin-vuepress/ui/config/write.js]: This is kind of a "hack". Trying to figure out a better way to handle this.

File: .gitignore [hunk 5]
```diff
@@ -5,3 +5,4 @@ node_modules
 TODOs.md
 packages/@vuepress/shared-utils/lib/
 types
+.vscode
```
Comment (ulivz) [inline .gitignore]: Do not commit `package-lock.json` since we use yarn.

**Pull request closed without merging**